### PR TITLE
fix(components): [notification] custom icon not work

### LINKED ISSUE #9497: [Component] [notification] [Bug] Custom icon on notification not displayed
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.15`
- Browser / OS: `Chrome/104.0.5112.79 Safari/537.36`
- Build Tool: `Vue CLI`

### Related Component

- `el-notification`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIHBsYWluIEBjbGljaz1cIm9wZW5cIj4gRGlzcGxheSBjdXN0b20gaWNvbiA8L2VsLWJ1dHRvbj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyBFbE5vdGlmaWNhdGlvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IGggfSBmcm9tICd2dWUnO1xuXG5jb25zdCBvcGVuID0gKCkgPT4ge1xuICBFbE5vdGlmaWNhdGlvbih7XG4gICAgdGl0bGU6ICdDdXN0b20gaWNvbicsXG4gICAgbWVzc2FnZTogJ1RoaXMgaXMgYSBtZXNzYWdlIHdpdGggY3VzdG9tIGljb24nLFxuICAgIGljb246IGgoJ2RpdicsIHsgaW5uZXJIVE1MOiAnSWMnIH0pLFxuICAgIGR1cmF0aW9uOiAwXG4gIH0pXG59XG48L3NjcmlwdD5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

### Steps to reproduce

Create a notification with a custom component for icon

### What is Expected?

Custom icon is displayed

### What is actually happening?

Custom icon is not displayed

### Additional comments

Seems occurs since 2.2.13, it worked on 2.2.12

### PR DESCRIPTION
**Pull request by holazz**
closes #9497

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by chenxch** [comment]
Missing default.

File: packages/components/notification/src/notification.ts [hunk 1]
```diff
@@ -24,8 +24,7 @@ export const notificationProps = buildProps({
     default: 4500,
   },
   icon: {
-    type: definePropType<string | Comment>([String, Object]),
-    default: '',
+    type: iconPropType,
```
Comment (chenxch) [inline packages/components/notification/src/notification.ts:27]: Missing default.

Comment (holazz) [reply packages/components/notification/src/notification.ts:27]: I think it's OK not to specify a default value, and I will fix the test case later. WDYT?

Comment (holazz) [reply packages/components/notification/src/notification.ts:27]: I created a PR to refactor `IconTypes`.

Comment (chenxch) [reply packages/components/notification/src/notification.ts:27]: Okey.

**Review by chenxch — APPROVED** [review]

**Review by ryuhangyeong — APPROVED** [review]

**Review by sxzz — APPROVED** [review]

**Pull request merged**